api/v11/api_v11_replication.go

```diff
@@ -41,23 +135,98 @@ func GetPolicyByName(
 	return &p.Policy[0], nil
 }
 
-func CreatePolicy(ctx context.Context, client api.Client, name string, sourcePath string, targetPath string, targetHost string, rpo int) error {
-	resp := ""
+func GetTargetPolicyByName(ctx context.Context, client api.Client, name string) (policy *TargetPolicy, err error) {
+	p := &TargetPolicies{}
+	err = client.Get(ctx, targetPoliciesPath, name, nil, nil, &p)
+	if err != nil || len(p.Policy) == 0 {
+		return nil, err
+	}
+	return &p.Policy[0], nil
+}
+
+func CreatePolicy(ctx context.Context, client api.Client, name string, sourcePath string, targetPath string, targetHost string, targetCert string, rpo int, enabled bool) error {
+	var policyResp Policy
 	body := &Policy{
 		Action:     "sync",
 		Id:         "",
 		Name:       name,
-		Enabled:    true,
+		Enabled:    enabled,
 		TargetPath: targetPath,
 		SourcePath: sourcePath,
 		TargetHost: targetHost,
 		JobDelay:   rpo,
+		TargetCert: targetCert,
 		Schedule:   "when-source-modified",
 	}
-	return client.Post(ctx, policiesPath, "", nil, nil, body, resp)
+	return client.Post(ctx, policiesPath, "", nil, nil, body, &policyResp)
 }
 
 func DeletePolicy(ctx context.Context, client api.Client, name string) error {
 	resp := ""
 	return client.Delete(ctx, policiesPath, name, nil, nil, &resp)
-}
\ No newline at end of file
+}
+
+func DeleteTargetPolicy(ctx context.Context, client api.Client, id string) error {
+	resp := ""
+	return client.Delete(ctx, targetPoliciesPath, id, nil, nil, &resp)
+}
+
+func UpdatePolicy(ctx context.Context, client api.Client, policy *Policy) error {
+	id := policy.Id
+	policy.Id = ""
+
+	return client.Put(ctx, policiesPath, id, nil, nil, policy, nil)
+}
+
+func ResetPolicy(ctx context.Context, client api.Client, name string) error {
+	resp := Policy{}
+	return client.Post(ctx, policiesPath, name+"/reset", nil, nil, nil, &resp)
+}
+
+func StartSyncIQJob(ctx context.Context, client api.Client, job *JobRequest) (*Job, error) {
+	var jobResp Job
+	return &jobResp, client.Post(ctx, jobsPath, "", nil, nil, job, &jobResp)
+}
+
+func GetReport(ctx context.Context, client api.Client, reportName string) (*Report, error) {
+	r := &Reports{}
+	err := client.Get(ctx, reportsPath, reportName, nil, nil, &r)
+	if err != nil || len(r.Reports) == 0 {
+		return nil, err
```

Comment:
Recommend to add change similar to the one in line 213 as in the second check err is nil.
if len(r.Reports) == 0 {
		return nil, fmt.Errorf("no reports found for policy %s", policyName)
	}

Reply:
Done